packages/react-router/modules/matchPath.js

```diff
@@ -26,11 +26,17 @@ const compilePath = (pattern, options) => {
 /**
  * Public API for matching a URL pathname to a path pattern.
  */
-const matchPath = (pathname, path, options = {}) => {
+const matchPath = (pathname, path, options = {}, parentMatch = null) => {
   const { exact = false, strict = false } = options
 
-  if (!path)
-    return { url: pathname, isExact: true, params: {} }
+  const absolute = isAbsolute(path)
+
+  if (path === undefined)
+    return { url: parentMatch ? parentMatch.url : '/', isExact: true, params: {} }
```

Comment:
Making `match.url` be `/` just for the sake of children resolving correctly leads me to believe there's a bug with the way the children resolve.